internal/rules/calver.go

```diff
@@ -0,0 +1,167 @@
+package rules
+
+import (
+	"strings"
+
+	"github.com/shipengqi/vc"
+
+	"github.com/orange-cloudavenue/kube-image-updater/internal/log"
+)
+
+type (
+	// calverMajor - The first number in the version.
+	calverMajor struct {
+		rule
+	}
+
+	// calverMinor - The second number in the version.
+	calverMinor struct {
+		rule
+	}
+
+	// calverPatch - The third and usually final number in the version. Sometimes referred to as the "micro" segment.
+	calverPatch struct {
+		rule
+	}
+
+	// calverModifier - The modifier is an optional part of the version.
+	calverModifier struct {
+		rule
+	}
+
+	// calverPrerelease - The prerelease is an optional part of the version.
+	calverPrerelease struct {
+		rule
+	}
+)
+
+func init() {
+	register(CalverMajor, &calverMajor{})
+	register(CalverMinor, &calverMinor{})
+	register(CalverPatch, &calverPatch{})
+	register(CalverModifier, &calverModifier{})
+	register(CalverPrerelease, &calverPrerelease{})
+}
+
+func (c *calverMajor) Evaluate() (matchWithRule bool, newTag string, err error) {
+	actualCV, err := vc.NewCalVerStr(c.actualTag)
+	if err != nil {
+		log.WithError(err).WithField("tag", c.actualTag).Error("Error parsing actual tag")
+		return false, "", err
+	}
+
+	for _, t := range c.tags {
+		cv, err := vc.NewCalVerStr(t)
+		if err != nil {
+			log.WithError(err).WithField("tag", t).Error("Error parsing tag")
+			continue
+		}
+
+		if cv.Major() > actualCV.Major() && actualCV.Prerelease() == "" && cv.Prerelease() == "" {
```

Comment:
Why not use Constraints ? 

https://github.com/shipengqi/vc#constraints
https://pkg.go.dev/github.com/shipengqi/vc@v0.2.0#NewConstraint